lib/devise/multi_email/models/confirmable.rb

```diff
@@ -38,64 +38,108 @@ module ConfirmableExtensions
 
         included do
           multi_email_association.include_module(EmailConfirmable)
-        end
-
-        # delegate before creating overriding methods
-        delegate :skip_confirmation!, :skip_confirmation_notification!, :skip_reconfirmation!, :confirmation_required?,
-                 :confirmation_token, :confirmed_at, :confirmation_sent_at, :confirm, :confirmed?, :unconfirmed_email,
-                 :reconfirmation_required?, :pending_reconfirmation?, to: :primary_email_record, allow_nil: true
-
-        # This need to be forwarded to the email that the user logged in with
-        def active_for_authentication?
-          login_email = current_login_email_record
 
-          if login_email && !login_email.primary?
-            super && login_email.active_for_authentication?
-          else
-            super
+          # Handle automatically confirming and linking `unconfirmed_email` if email change is not postponed
+          before_update do
+            if !postponed_email_change? && multi_email.unconfirmed_email_changes?
+              multi_email.set_primary_record_to(
+                multi_email.unconfirmed_email_record,
+                skip_confirmations: true
+              )
+            end
           end
-        end
-
-        # Shows email not confirmed instead of account inactive when the email that user used to login is not confirmed
-        def inactive_message
-          login_email = current_login_email_record
 
-          if login_email && !login_email.primary? && !login_email.confirmed?
-            :unconfirmed
-          else
-            super
+          # Don't send notifications when the email records are saved
+          # when saving the parent record.
+          if respond_to?(:after_commit)
+            after_commit(prepend: true){ multi_email.primary_email_record.try(:skip_confirmation_notification!) }
```

Comment:
@allenwq this is similarly named to `skip_confirmation!` but `skip_confirmation_notification!` is actually just to skip sending the email notification. It keeps the confirmation status as it was.

When the email address is changed, Devise stores the changed email in `unconfirmed_email`. Relying on the Devise lifecycle, it will still send a confirmation email to the new email address that was specified. This callback simply prevents sending a notification email for all other unconfirmed email addresses. As a matter of fact, without this callback, the confirmation for the new "unconfirmed" email will be sent twice.